hadoop-hdds/rocks-native/src/main/patches/rocks-native.patch

```diff
@@ -16,592 +16,557 @@
++#include "file/writable_file_writer.h"
++#include "options/cf_options.h"
++
++namespace ROCKSDB_NAMESPACE {
++
++struct RawSstFileReader::Rep {
++  Options options;
++  EnvOptions soptions_;
++  ReadOptions read_options_;
++  ImmutableOptions ioptions_;
++  MutableCFOptions moptions_;
++  InternalKeyComparator internal_comparator_;
++  std::unique_ptr<TableProperties> table_properties_;
++  std::unique_ptr<TableReader> table_reader_;
++  std::unique_ptr<RandomAccessFileReader> file_;
++
++  Rep(const Options& opts, bool verify_checksum, size_t readahead_size)
++      : options(opts),
++        soptions_(EnvOptions()),
++        read_options_(verify_checksum, false),
++        ioptions_(options),
++        moptions_(ColumnFamilyOptions(options)),
++        internal_comparator_(InternalKeyComparator(BytewiseComparator())) {
++    read_options_.readahead_size = readahead_size;
++  }
++};
++
++RawSstFileReader::RawSstFileReader(const Options& options,
++                                   const std::string& file_name,
++                                   size_t readahead_size,
++                                   bool verify_checksum,
++                                   bool silent) :rep_(new Rep(options,
++                                                               verify_checksum,
++                                                               readahead_size)) {
++  file_name_ = file_name;
++  silent_ = silent;
++  options_ = options;
++  file_temp_ = Temperature::kUnknown;
++  init_result_ = GetTableReader(file_name_);
++}
++
++RawSstFileReader::~RawSstFileReader() {}
++
++
++
++extern const uint64_t kBlockBasedTableMagicNumber;
++extern const uint64_t kLegacyBlockBasedTableMagicNumber;
++extern const uint64_t kPlainTableMagicNumber;
++extern const uint64_t kLegacyPlainTableMagicNumber;
++
++Status RawSstFileReader::GetTableReader(const std::string& file_path) {
++  // Warning about 'magic_number' being uninitialized shows up only in UBsan
++  // builds. Though access is guarded by 's.ok()' checks, fix the issue to
++  // avoid any warnings.
++  uint64_t magic_number = Footer::kNullTableMagicNumber;
++
++  // read table magic number
++  Footer footer;
++
++  const auto& fs = options_.env->GetFileSystem();
++  std::unique_ptr<FSRandomAccessFile> file;
++  uint64_t file_size = 0;
++  FileOptions fopts = rep_->soptions_;
++  fopts.temperature = file_temp_;
++  Status s = fs->NewRandomAccessFile(file_path, fopts, &file, nullptr);
++  if (s.ok()) {
++    s = fs->GetFileSize(file_path, IOOptions(), &file_size, nullptr);
++  }
++
++  // check empty file
++  // if true, skip further processing of this file
++  if (file_size == 0) {
++    return Status::Aborted(file_path, "Empty file");
++  }
++
++  rep_->file_.reset(new RandomAccessFileReader(std::move(file), file_path));
++
++  FilePrefetchBuffer prefetch_buffer(
++      0 /* readahead_size */, 0 /* max_readahead_size */, true /* enable */,
++      false /* track_min_offset */);
```

Comment:
Shouldn't we be using `readahead_size` passed as class constructor?

Reply:
this is a different readahead_size we want to just read the prefetch buffer to get the magic number here. No read ahead buffers needed here.